### PR TITLE
Install splign v2.1.0

### PR DESCRIPTION
**Pull request by jayramr**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/splign/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+requirements:
+  run:
+    - bzip2
+    - lmdb
+    - libuv
+    - pcre
+    - libnghttp2
+    - sqlite
+    - bz2file
```
Comment (marcelm) [inline recipes/splign/meta.yaml:26]: Please remove this dependency: `bz2file` is a Python package that wraps bzip2. The correct dependency is just `bzip2`.

**Comment by marcelm** [comment]
This is binary repackaging, which can be a bit problematic. This is the error that you are getting:
```
splign: error while loading shared libraries: libbz2.so.1: cannot open shared object file: No such file or directory
```
In short, `splign` incorrectly links against a dynamic library that is not actually provided by `bzip2`. `bzip2` only provides `libbz2.so` and `libbz2.so.1.0`. See some [more details](https://github.com/conda-forge/bzip2-feedstock/pull/6).

Because Debian’s `bzip2` package *does* provide the `libbz2.so.1` name in its `bzip2` package (that’s a choice the Debian maintainers made), `splign` runs if the `bzip2` package happens to be available on the system (it doesn’t use the Conda-provided library). You get the test failure from within mulled-build, which is in a container where *only* Conda packages are available, so the system library cannot be found.

The preferred method of fixing this would be to compile from sources, but I know from experience that these NCBI tools are extraordinarily annoying to get to compile. Alternatively, you can use `patchelf` to modify the existing binaries:
```
patchelf --replace-needed libbz2.so.1 libbz2.so.1.0 splign
```
Do the same for `compart` and add a test for `compart` (unless you do not want to package it, in which case you should ensure you only install the `splign` binary).

**Comment by jayramr** [comment]
Thanks for your pointer , i haven't tried with patchelf tool. Could you help me how can I compile using patchelf ?

**Comment by marcelm** [comment]
Do you want to test it locally? On Ubuntu, I just did `sudo apt install patchelf`. You can also use `conda install patchelf`.

**Comment by jayramr** [comment]
> Do you want to test it locally? On Ubuntu, I just did `sudo apt install patchelf`. You can also use `conda install patchelf`.

Sorry what I meant how can I use this tool "patchelf" to update the library name from "libbz2.so.1 libbz2.so.1.0" ?Do you have any example to show, so that I can refer that 

Thanks

**Comment by marcelm** [comment]
I wrote that above, just use this command in `build.sh`:
```
patchelf --replace-needed libbz2.so.1 libbz2.so.1.0 splign
```

**Comment by jayramr** [comment]
> I wrote that above.
> 
> > Alternatively, you can use patchelf to modify the existing binaries:
> > ```
> > patchelf --replace-needed libbz2.so.1 libbz2.so.1.0 splign
> > ```
> 
> Just use that command in `build.sh`.

Ah i see thanks let me try :)

**Pull request closed without merging**